src/transport_portals4.c

```diff
@@ -720,25 +722,22 @@ shmem_transport_startup(void)
 int
 shmem_transport_fini(void)
 {
-    size_t i;
-
     /* synchronize the atomic cache, if there is one */
     shmem_transport_syncmem();
 
-    /* Free all contexts.  This performs a quiet on each context, ensuring all
-     * operations have completed before proceeding with shutdown. */
+    /* Free all contexts on the world team.  This performs a quiet on each context,
+     * ensuring all operations have completed before proceeding with shutdown. */
 
-    for (i = 0; i < shmem_transport_portals4_contexts_len; ++i) {
-        if (shmem_transport_portals4_contexts[i]) {
-            if (shmem_transport_portals4_contexts[i]->options & SHMEM_CTX_PRIVATE)
+    for (size_t i = 0; i < shmem_internal_team_world.contexts_len; ++i) {
+        if (shmem_internal_team_world.contexts[i]) {
+            if (shmem_internal_team_world.contexts[i]->options & SHMEM_CTX_PRIVATE)
                 RAISE_WARN_MSG("Shutting down with unfreed private context (%zu)\n", i);
-            shmem_transport_quiet(shmem_transport_portals4_contexts[i]);
-            shmem_transport_ctx_destroy(shmem_transport_portals4_contexts[i]);
+            shmem_transport_quiet(shmem_internal_team_world.contexts[i]);
+            shmem_transport_ctx_destroy(shmem_internal_team_world.contexts[i]);
         }
     }
 
-    if (shmem_transport_portals4_contexts)
-        free(shmem_transport_portals4_contexts);
+    if (shmem_internal_team_world.contexts) free(shmem_internal_team_world.contexts);
 
```

Comment:
Missing destroy of shared team.  Can we call `shmem_internal_team_destroy` rather than doing this manually here?  Also, let me know if you need help with testing the Portals build.

Reply:
Same question/concern as in OFI...

And yes, I could definitely use help with testing portals - thanks!

Reply:
With the one-line bugfix https://github.com/davidozog/sandia-shmem/pull/8 all portals tests pass!  💯

Reply:
Whoops, forgot portals again.  Does https://github.com/Sandia-OpenSHMEM/SOS/pull/886/commits/7c57f6f8a8d657ebbd013beba54974ae1e272e2b work?